src/modules/roc_packet/target_stdio/roc_packet/ip_to_str.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_packet/ip_to_str.h"
+#include "roc_core/log.h"
+
+namespace roc {
+namespace packet {
+
+ip_to_str::ip_to_str(const Address& addr) {
+    buffer_[0] = '\0';
+
+    switch (addr.version()) {
+    case 4: {
+        if (!addr.get_ip(buffer_, sizeof(buffer_))) {
+            roc_log(LogError, "ip_str_formatter: can't format ip");
+        }
+
+        break;
+    }
+    case 6: {
+        buffer_[0] = '[';
+
+        if (!addr.get_ip(buffer_ + 1, sizeof(buffer_) - 1)) {
```

Comment:
Oh, by the way shouldn't it be `sizeof(buffer_) - 2` instead of `sizeof(buffer_) - 1`? Because of the closing `]`.

Reply:
Previously we specified that we have `sizeof(buffer) - 1` bytes and expected that 2snprintf` would properly format the remaining part.

Agree, that in my implementation in some very rare cases we won't have enough space in the underlying buffer for the remaining `]`.

Reply:
Also, it does not a big deal but for the ip part we can use a smaller buffer because the string representation of the ipv6 address does not exceed 64 bytes. On Linux we have `INET6_ADDRSTRLEN`.

Reply:
> On Linux we have INET6_ADDRSTRLEN.

We should avoid using it here because this will make ip_to_str platform-dependent.